src/mlpack/methods/gmm/em_fit_impl.hpp

```diff
@@ -91,31 +92,36 @@ Estimate(const arma::mat& observations,
     {
       // Store conditional probabilities into condProb vector for each
       // Gaussian.  First we make an alias of the condProb vector.
-      arma::vec condProbAlias = condProb.unsafe_col(i);
-      dists[i].Probability(observations, condProbAlias);
-      condProbAlias *= weights[i];
+      arma::vec condLogProbAlias = condLogProb.unsafe_col(i);
+      dists[i].LogProbability(observations, condLogProbAlias);
+      condLogProbAlias += log(weights[i]);
     }
 
     // Normalize row-wise.
-    for (size_t i = 0; i < condProb.n_rows; i++)
+    for (size_t i = 0; i < condLogProb.n_rows; i++)
     {
       // Avoid dividing by zero; if the probability for everything is 0, we
       // don't want to make it NaN.
-      const double probSum = accu(condProb.row(i));
-      if (probSum != 0.0)
-        condProb.row(i) /= probSum;
+      const double probSum = mlpack::math::AccuLog(condLogProb.row(i));
+      if (probSum != -std::numeric_limits<double>::infinity())
+        condLogProb.row(i) -= probSum;
     }
 
     // Store the sum of the probability of each state over all the observations.
-    arma::vec probRowSums = trans(arma::sum(condProb, 0 /* columnwise */));
+    arma::vec probRowSums(dists.size());
+    for (size_t i = 0; i < dists.size(); ++i)
+    {
+        probRowSums(i) = mlpack::math::AccuLog(condLogProb.col(i));
```

Comment:
```suggestion
      probRowSums(i) = mlpack::math::AccuLog(condLogProb.col(i));
```

The indentation should only be two spaces here. :)